packages/gatsby/src/redux/persist.ts

```diff
@@ -86,6 +86,13 @@ function guessSafeChunkSize(values: [string, IGatsbyNode][]): number {
     maxSize = Math.max(size, maxSize)
   }
 
+  // Sends a warning once if any of the chunkSizes exceeds approx 500kb limit
+  if (maxSize > 500000) {
+    report.warn(
+      `The size of least one page context exceeded 500kb, this could lead to degraded performance. Consider putting less data in the page context.`
```

Comment:
```suggestion
      `The size of at least one page context chunk exceeded 500kb, which could lead to degraded performance. Consider putting less data in the page context.`
```

This could use a couple of tweaks to the grammar but I think it’s a great addition! As a suggestion, I’ve also added the word “chunk” in an attempt to clarify what’s being measured.